src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,104 @@
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        user.setLogin(INVALID_LOGIN);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Invalid login: must be at least 6 characters long.",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        user.setLogin(null);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Invalid login: cannot be null or empty.",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_invalidPassword_notOk() {
+        user.setPassword(INVALID_PASSWORD);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Invalid password: must be at least 6 characters long.",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        user.setPassword(null);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Invalid password: cannot be null or empty.",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_sameLogin_notOk() {
+        storage.add(user);
+        registrationService.register(user);
+        assertTrue(Storage.people.contains(user),
+                "User with the same login should not exist");
+    }
```

This test does not make any sense, we assert that we have some user in the storage(there are actually 2 identical users in it), and if it does not - we fail the test with a message that it shouldn't exist,

This `register()` call should throw an exception which we should assert here because we can not allow `storage` to have 2 users with the same login as per our requirements. Please be more careful about the task description next time because this logic is missing in the service.